### PR TITLE
Proper buckling

### PR DESCRIPTION
**Pull request by LagannPM**

### CODE REVIEW

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -58,31 +58,67 @@
...
-				SPAN_DANGER("You are buckled to [src] by [user.name]!"),\
-				SPAN_NOTICE("You hear metal clanking."))
+	
+	
+	var/buckleTime = 25
+	if (M == user)
+		user.set_dir(src.dir)
+		buckleTime = 8
+		user.visible_message(\
+		SPAN_NOTICE("[user] attempts to buckle themselves into \the [src]!</span>"),\
```
Comment (NanakoAC) [inline code/game/objects/buckling.dm]: This message is unnecessary noise, someone trying to buckle themselves is not important for everyone to see.

Comment (LagannPM) [reply code/game/objects/buckling.dm]: deleted this

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -58,31 +58,67 @@
...
+		user.visible_message(\
+		SPAN_NOTICE("[user] attempts to buckle [M] into \the [src]!</span>"),\
+		SPAN_NOTICE("You attempt to buckle [M] into \the [src]!</span>"),\
+		SPAN_NOTICE("You hear metal clanking."))
+	
+	if(do_after(user, buckleTime, src))
+		if(buckle_mob(M))
+			playsound(src.loc, 'sound/machines/Custom_closetopen.ogg', 15, 1, -3)
+			if(M == user)
+				M.visible_message(\
```
[the remark this answers is not in the document]
Comment (LagannPM) [reply code/game/objects/buckling.dm]: came with a compromise, only targets in adjacent tiles receive messages

File: code/game/objects/buckling.dm [hunk 3]
```diff
@@ -58,31 +58,67 @@
...
+		if(M)
+			if(user.loc != M.loc)
+				user.do_attack_animation(src)
+			if(M != user)
+				M.visible_message(\
+					SPAN_NOTICE("[M.name] was unbuckled by [user.name]!"),\
+					SPAN_NOTICE("You were unbuckled from [src] by [user.name]."),\
+					SPAN_NOTICE("You hear metal clanking."))
+			else
+				M.visible_message(\
```
[the remark this answers is not in the document]
Comment (LagannPM) [reply code/game/objects/buckling.dm]: same here

**Pull request merged**